django_mongodb/fields/embedded_model.py

```diff
@@ -0,0 +1,165 @@
+        elif isinstance(value, dict):
+            embedded_model = self.stored_model(value)
+            attribute_values = value
+        else:
+            return value
+        # Pass values through respective fields' to_python(), leaving
+        # fields for which no value is specified uninitialized.
+        attribute_values = {
+            field.attname: field.to_python(attribute_values[field.attname])
+            for field in embedded_model._meta.fields
+            if field.attname in attribute_values
+        }
+        # Create the model instance.
+        instance = embedded_model(**attribute_values)
+        instance._state.adding = False
+        return instance
+
+    def get_db_prep_save(self, embedded_instance, connection):
+        """
+        Apply pre_save() and get_db_prep_save() of embedded instance
+        fields and passes a field => value mapping down to database
+        type conversions.
+
+        The embedded instance will be saved as a column => value dict
+        in the end (possibly augmented with info about instance's model
+        for untyped embedding), but because we need to apply database
+        type conversions on embedded instance fields' values and for
+        these we need to know fields those values come from, we need to
+        entrust the database layer with creating the dict.
+        """
+        if embedded_instance is None:
+            return None
+        # The field's value should be an instance of the model given in
+        # its declaration or at least of some model.
+        embedded_model = self.embedded_model or models.Model
+        if not isinstance(embedded_instance, embedded_model):
+            raise TypeError(
+                f"Expected instance of type {embedded_model!r}, not {type(embedded_instance)!r}."
+            )
+        # Apply pre_save() and get_db_prep_save() of embedded instance
+        # fields, create the field => value mapping to be passed to
+        # storage preprocessing.
+        field_values = {}
+        add = embedded_instance._state.adding
+        for field in embedded_instance._meta.fields:
+            value = field.get_db_prep_save(
+                field.pre_save(embedded_instance, add), connection=connection
+            )
+            # Exclude unset primary keys (e.g. {'id': None}).
+            if field.primary_key and value is None:
+                continue
+            field_values[field.attname] = value
+        # Let untyped fields store model info alongside values.
+        # Use fake RawFields for additional values to avoid passing
+        # embedded_instance to database conversions and to give
+        # backends a chance to apply generic conversions.
+        if self.embedded_model is None:
+            field_values.update(
+                (
+                    ("_module", embedded_instance.__class__.__module__),
+                    ("_model", embedded_instance.__class__.__name__),
+                )
+            )
```

Comment:
When you have an `EmbeddedModelField()` that doesn't specify a model (i.e. can store any model), then you need to store the model info in the database for converting it back after you store it. See [Generic Embedding](https://django-mongodb-engine.readthedocs.io/en/latest/topics/embedded-models.html#generic-embedding). Do you think it's important to support? It doesn't seem practical that we could do schema changes for such cases since ANY model field change might have to be propagated into all such fields. Querying into such fields also seems a bit questionable since the fields will be different among different models.

We could remove this feature to simplify scope for now and add it back if there's demand. I'm not sure there's any other code to support this besides:
```diff
index 3b72ef3..525f0a0 100644
--- a/django_mongodb/fields/embedded_model.py
+++ b/django_mongodb/fields/embedded_model.py
@@ -57,20 +57,12 @@ class EmbeddedModelField(models.Field):
     def stored_model(self, column_values):
         """
         Returns the fixed embedded_model this field was initialized
-        with (typed embedding) or tries to determine the model from
-        _module / _model keys stored together with column_values
-        (untyped embedding).
+        with (typed embedding).
 
         We give precedence to the field's definition model, as silently
         using a differing serialized one could hide some data integrity
         problems.
-
-        Note that a single untyped EmbeddedModelField may process
-        instances of different models (especially when used as a type
-        of a collection field).
         """
-        module = column_values.pop("_module", None)
-        model = column_values.pop("_model", None)
         if self.embedded_model is not None:
             return self.embedded_model
         if module is not None:
@@ -149,14 +141,6 @@ class EmbeddedModelField(models.Field):
             if field.primary_key and value is None:
                 continue
             field_values[field.attname] = value
-        if self.embedded_model is None:
-            # Untyped fields must store model info alongside values.
-            field_values.update(
-                (
-                    ("_module", embedded_instance.__class__.__module__),
-                    ("_model", embedded_instance.__class__.__name__),
-                )
-            )
         # This instance will exist in the database soon.
         # TODO.XXX: Ensure that this doesn't cause race conditions.
         embedded_instance._state.adding = False
```